apollo-router/src/executable.rs

```diff
@@ -228,6 +230,10 @@ pub struct Opt {
     /// Display version and exit.
     #[clap(action = ArgAction::SetTrue, long, short = 'V')]
     pub(crate) version: bool,
+
+    /// Override the default address to listen on for only the supergraph.
+    #[clap(long = "listen")]
+    listen_address: Option<SocketAddr>,
```

Comment:
I think it might be a good idea to make this pattern easily extendable to other configuration options we may want to add in the future. 

What do you think about separating this out into its own struct like this

```rust
pub struct OverrideYamlOpts {
  #[clap(long = "listen")]
  listen_address: Option<SocketAddr>
}
```

and then the existing line can be this:

```rust
#[clap(flatten)]
override_yaml_opts: OverrideYamlOpts
```

In the future, if there are other options that override entries in the YAML file, they can be added here and handled similarly elsewhere in the code.

Reply:
That could help, but I'd worry about adding too many cli args. If the core problem is that it's cumbersome for rover to change the router config, that's the point we should address